### DISCUSSION TITLE
Add Network & Socket to Feather RP2040

### DISCUSSION THREAD

**Discussion by SquirtleSquadLeader** [micropython/micropython#10729] [General]
I am relatively new to MicroPython.

I would like to use an Ethernet Featherwing with my RP2040 Feather.  Is there an easy way to add this functionality?

Any advice is welcomed (I appreciate being pointed to guides if any exits).  Thanks in advance.

EDIT: Should I be heading down this path https://docs.micropython.org/en/latest/develop/natmod.html#supported-features-and-limitations ?

**Comment by robert-hh** [comment]
It should be easier. The Feather wing uses a W5500 controller. So you can take an existing board with W5500, like the W5500_EVB_PICo as a template and adapt the board files, like mpconfigboard.h and mpconfigboard.cmake. You have as well wire the Reset and IRQ pin at JP1 of the feather wing to GPIO pins of the RP2040 (at least reset).
Then you can build your own formware,

**Reply by SquirtleSquadLeader** [reply]
Hmm Okay, I'll need to do some research.  This is definitely not something I have any familiarity with.  I'm trying to migrate over from Circuit Python - so I still haven't tackled any customization past drag and drop modules.
